src/osd/osd_types.cc

```diff
@@ -5878,3 +5878,33 @@ ostream& operator<<(ostream& out, const store_statfs_t &s)
       << ")";
   return out;
 }
+
+void OSDOp::clear_data(vector<OSDOp>& ops)
+{
+  for (unsigned i = 0; i < ops.size(); i++) {
+    OSDOp& op = ops[i];
+    op.outdata.clear();
+    if (ceph_osd_op_type_attr(op.op.op) &&
+        op.op.xattr.name_len &&
+	op.indata.length() >= op.op.xattr.name_len) {
+      bufferptr bp(op.op.xattr.name_len);
+      bufferlist bl;
+      bl.append(bp);
+      bl.copy_in(0, op.op.xattr.name_len, op.indata);
+      op.indata.claim(bl);
```

Comment:
got it, `Message::print()` is used to print out the "description" field. but my question of

> and why you want to rebuild `op.indata`? BTW, what you accomplish here is equivalent to what `ceph::buffer::list::rebuild()` does, i think.

still holds. `bufferlist::write(int off, int len, std::ostream& out)` is able to handle non-contiguous bufferlist.

and yeah, i am a little bit concern about the performance impact of this change. this change

1. keeps the payload around for awhile, and
1. rebuilds the `op.indata` (which i don't think is necessary, though)